### PR TITLE
Add rules for interaction between new language features a older language versions.

### PR DESCRIPTION
**Pull request by lrhn**

### CODE REVIEW

File: accepted/future-releases/small-features-21Q1/feature-specification.md [hunk 1]
```diff
@@ -113,3 +113,30 @@ on the `int` class, which will work similarly to `>>`, but will zero-extend inst
...
 
 Backends may want to optimize this to use available bitwise shift operations (like `>>>` in JavaScript), and intrinsify the function if possible. This can be done at any later time, though.
+
+## Mixed Mode Programs
+
+Libraries using a language version prior to the introduction of these features (opted out libraries) 
+interact with libraries using those features (opted in libraries) as follows.
+
+In an opted out library:
+* It is a compile-time error to declare an operator method named `>>>`, or if to have `e1 >>> e2` as an expression.
```
Comment (eernstg) [inline accepted/future-releases/small-features-21Q1/feature-specification.md:123]: Typo: `or if` --> `or`.

File: accepted/future-releases/small-features-21Q1/feature-specification.md [hunk 2]
```diff
@@ -113,3 +113,30 @@ on the `int` class, which will work similarly to `>>`, but will zero-extend inst
...
+* It is a compile-time error to contain an explicit type argument on an annotation constructor invocation.
```
Comment (eernstg) [inline accepted/future-releases/small-features-21Q1/feature-specification.md:124]: The meaning of the word `contain` isn't obvious here. Maybe `..error if a type argument list occurs..`?

File: accepted/future-releases/small-features-21Q1/feature-specification.md [hunk 3]
```diff
@@ -113,3 +113,30 @@ on the `int` class, which will work similarly to `>>`, but will zero-extend inst
...
+* It is a compile-time error to declare a type parameter with a generic function type (GFT) as bound.
+* It is a compile-time to use a GFT as a type argument anywhere. This includes:
```
Comment (eernstg) [inline accepted/future-releases/small-features-21Q1/feature-specification.md:126]: Typo: `compile-time` --> `compile-time error`.

File: accepted/future-releases/small-features-21Q1/feature-specification.md [hunk 4]
```diff
@@ -113,3 +113,30 @@ on the `int` class, which will work similarly to `>>`, but will zero-extend inst
...
+  (That is, it's not an error to simply have an expression with a static type which includes GFT as a type argument or bound.)
+* It is *not* an error to export a symbol from an opted in library which uses a GFT as a type argument or bound.
+
+That is, there is no new expressiveness in an opted out library due to these features. 
+Everything which was previously an error to write, explicitly or implicity, is still an error.
+
+This does mean, since type parameter bounds are invariant, that if a class in an opted in library declared an instance
+member with a GFT-bounded type parameter, an opted out library cannot implement that interface.
+No existing classes declares such a boundo, and changing the bound is a breaking change no matter what it's changed to, 
+so that is not expected to be am issue. We have no plans to add GFT-bounds to existing platform library interfaces.
```
Comment (eernstg) [inline accepted/future-releases/small-features-21Q1/feature-specification.md:142]: Typo: `am` --> `an`.

File: accepted/future-releases/small-features-21Q1/feature-specification.md [hunk 5]
```diff
@@ -113,3 +113,30 @@ on the `int` class, which will work similarly to `>>`, but will zero-extend inst
...
+    * Types produced by expanding references to type aliases into their aliased type.
+    * The corresponding explicit extension invocation for an implicit extension invocation.
+* It is *not* an error to refer to or use a symbol from an opted in library which uses a GFT as a type argument or bound.
+  (That is, it's not an error to simply have an expression with a static type which includes GFT as a type argument or bound.)
+* It is *not* an error to export a symbol from an opted in library which uses a GFT as a type argument or bound.
+
+That is, there is no new expressiveness in an opted out library due to these features. 
+Everything which was previously an error to write, explicitly or implicity, is still an error.
+
+This does mean, since type parameter bounds are invariant, that if a class in an opted in library declared an instance
```
Comment (eernstg) [inline accepted/future-releases/small-features-21Q1/feature-specification.md:139]: Might as well say `declares`.

File: accepted/future-releases/small-features-21Q1/feature-specification.md [hunk 6]
```diff
@@ -113,3 +113,30 @@ on the `int` class, which will work similarly to `>>`, but will zero-extend inst
...
+member with a GFT-bounded type parameter, an opted out library cannot implement that interface.
+No existing classes declares such a boundo, and changing the bound is a breaking change no matter what it's changed to,
```
Comment (eernstg) [inline accepted/future-releases/small-features-21Q1/feature-specification.md:141]: Typo: `boundo` --> `bound`.

**Review by eernstg — APPROVED** [review]
LGTM, noting a few typos.

**Review by leafpetersen — APPROVED** [review]
LGTM, thanks!

**Pull request merged**